### PR TITLE
Ports duplicated slipping code to a component

### PR DESCRIPTION
**Pull request by Cyberboss**
Instead of having Crossed overrides for all of them that do the same thing. Just AddComponent() and boom, slippery when Crossed.

Also minor change to base code, added another proc called on the component itself after ComponentActivated is called on the parent

### CODE REVIEW

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 1]
```diff
@@ -29,6 +29,9 @@
 	metal = ALUMINUM_FOAM
 	icon_state = "mfoam"
 
+/obj/effect/particle_effect/foam/metal/MakeSlippery()
+	return
```
Comment (optimumtact) [inline code/game/objects/effects/effect_system/effects_foam.dm]: what is this

Comment (optimumtact) [inline code/game/objects/effects/effect_system/effects_foam.dm]: oh I see, hmm, can this be a var or flag instead

**Comment by ChangelingRain** [comment]
Do this for d4s, glass shards, and cacti too.

**Comment by Cyberboss** [comment]
Different PR but sure

File: code/game/objects/effects/effect_system/effects_foam.dm (hunk 1 above)
Comment (Jalleo) [inline code/game/objects/effects/effect_system/effects_foam.dm]: Possibly I dont know the best method.

**Comment by Cyberboss** [comment]
DNM until #29647 is merged

**Comment by optimumtact** [comment]
:point_up:

File: code/datums/components/slippery.dm [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	var/mob/slip_victim
+
+/datum/component/slippery/New(datum/P, _intensity, _lube_flags = NONE)
+	..()
+	intensity = max(_intensity, 0)
+	lube_flags = _lube_flags
+	if(ismovableatom(P))
+		RegisterSignal(COMSIG_MOVABLE_CROSSED, .proc/Slip)
+	else
+		RegisterSignal(COMSIG_ATOM_ENTERED, .proc/Slip)
```
Comment (AnturK) [inline code/datums/components/slippery.dm]: Technically you can have movables that slip on entered but it's not a thing right now so this is fine.

File: code/game/objects/effects/effect_system/effects_foam.dm (hunk 1 above)
Comment (AnturK) [inline code/game/objects/effects/effect_system/effects_foam.dm]: This is fine in my opinion.

File: code/game/objects/items/devices/PDA/PDA_types.dm [hunk 3]
```diff
@@ -7,16 +7,20 @@
...
+	AddComponent(/datum/component/slippery, 120, NO_SLIP_WHEN_WALKING)
+
+/obj/item/device/pda/clown/ComponentActivated(datum/component/C)
+	..()
+	var/datum/component/slippery/S = C
+	if(!istype(S))
+		return
+	var/mob/living/carbon/human/M = S.slip_victim
+	if (istype(M) && (M.real_name != src.owner))
+		var/obj/item/weapon/cartridge/virus/cart = cartridge
```
Comment (AnturK) [inline code/game/objects/items/devices/PDA/PDA_types.dm]: You didn't use clown cartridge type, let's not give clowns reloadable detomax.

File: code/game/turfs/open.dm [hunk 4]
```diff
@@ -216,6 +217,33 @@
...
+
+/turf/open/ComponentActivated(datum/component/C)
+	..()
+	var/datum/component/slippery/S = C
+	if(!istype(S))
+		return
+	var/mob/living/L = S.slip_victim
+	switch(wet)
+		if(TURF_WET_WATER)
+			L.inertia_dir = 0
```
Comment (AnturK) [inline code/game/turfs/open.dm]: Isn't this the part of the core slip function ? Why are these here ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/turfs/open.dm (hunk 4 above)
Comment (Cyberboss) [inline code/game/turfs/open.dm]: Actually, it only triggers if the slip was specifically because of TURF_WET_WATER and no other thing does this.

Comment (AnturK) [inline code/game/turfs/open.dm]: What's the point of this anyway, you can't slip in no gravity from turfs can you?

**Review by AnturK — APPROVED** [review]

File: code/game/turfs/open.dm (hunk 4 above)
Comment (Cyberboss) [inline code/game/turfs/open.dm]: literally no clue, want me to remove it?
